### ISSUE TITLE
Disabling LEDs

### ISSUE THREAD

**Issue by GermanCoding** [cyrozap/jmb58x-re#2]
Hi,

I own a JMB585-based board which unfortunately features way too bright green LEDs. So I was looking for possible options to disable them, as I don't really need them. Of course just cutting PCB traces or the LEDs themselves would probably be the easiest here, but destructive methods are always a bit scary to me. So I looked into whether there are possible software ways to turn them off.

Based on your notes here it looks like these LEDs are probably driven by GPIO pins on the chip, which could perhaps be disabled via a configuration change? Can you confirm this? Do you have made progress with the instruction/configuration reversing perhaps?

Though even if so, there's probably not an easy way to reflash the flash with a new configuration, right? I'm a beginner when it comes to messing with hardware, so I'm not sure if I want to attempt desoldering and then resoldering a flash chip (Though I do have friends which may be able to assist me with such endeavours).

Interested to hear your thoughts & many thanks for the notes - really helpful!

Max

**Comment by cyrozap** [comment]
Yes, I can confirm that the LEDs can be disabled by a configuration change, though I'm not sure exactly how to do that. At one point I tried writing the flash of one card (full-size PCIe) to the flash of another card (M.2), and upon powering it on I noticed that some of the LEDs were out. After writing the original flash contents back to that card, all the LEDs worked the same as they did before. So while I'm certain that the configuration sequence is what defines which LED signal gets routed to what GPIO pin and the pin modes of the GPIO pins, I haven't done enough RE to determine exactly which bits control what pins and signals.

Regarding flashing, I still don't know how to do that over PCIe, or if that's even possible at all. On many boards that use this chip there's what looks to be an in-circuit programming header whose traces connect to the JMB585 IC (not the flash), but I haven't yet figured out how to use that. So for now, the only way to (reliably) rewrite the flash is to desolder it and program it externally.

All that said, I think the simplest and least-destructive way to solve your problem would be to either cover the LEDs with electrical tape (or any other opaque, non-conductive tape or sticker) or to use a black permanent marker to try and black out the LEDs (or at least significantly reduce their brightness). These methods are rather low-tech, but they should solve your problem without requiring you to do any soldering.
